src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -70,45 +71,74 @@ class F extends ValidationTest {
+    const extension: GPUExtensionName = kAllTextureFormatInfo[format].extension;
+    await t.selectDeviceOrSkipTestCase({ extensions: [extension] });
+
+    t.device.createTexture(descriptor);
   });
 
-g.test('validation_of_mipLevelCount')
+g.test('mipLevelCount_validation_on_dimensions')
   .params([
-    { width: 32, height: 32, mipLevelCount: 1, _success: true }, // mipLevelCount of 1 is allowed
     { width: 32, height: 32, mipLevelCount: 0, _success: false }, // mipLevelCount of 0 is not allowed
+    { width: 32, height: 32, mipLevelCount: 1, _success: true }, // mipLevelCount of 1 is allowed
+    { width: 32, height: 32, mipLevelCount: 3, _success: true }, // non full mip chains are allowed (Mip leve sized: 32, 16, 8)
     { width: 32, height: 32, mipLevelCount: 6, _success: true }, // full mip chains are allowed (Mip level sizes: 32, 16, 8, 4, 2, 1)
     { width: 31, height: 32, mipLevelCount: 6, _success: true }, // full mip chains are allowed (Mip level sizes: 31x32, 15x16, 7x8, 3x4, 1x2, 1x1)
     { width: 32, height: 31, mipLevelCount: 6, _success: true }, // full mip chains are allowed (Mip level sizes: 32x31, 16x15, 8x7, 4x3, 2x1, 1x1)
     { width: 31, height: 32, mipLevelCount: 7, _success: false }, // too big mip chains on width are disallowed (Mip level sizes: 31x32, 15x16, 7x8, 3x4, 1x2, 1x1, ?x?)
     { width: 32, height: 31, mipLevelCount: 7, _success: false }, // too big mip chains on height are disallowed (Mip level sizes: 32x31, 16x15, 8x7, 4x3, 2x1, 1x1, ?x?)
     { width: 32, height: 32, mipLevelCount: 100, _success: false }, // undefined shift check if miplevel is bigger than the integer bit width
     { width: 32, height: 8, mipLevelCount: 6, _success: true }, // non square mip map halves the resolution until a 1x1 dimension. (Mip maps: 32 * 8, 16 * 4, 8 * 2, 4 * 1, 2 * 1, 1 * 1)
+    { width: 32, height: 32, arrayLayerCount: 64, mipLevelCount: 7, _success: false }, // array layer count for 2D texture should not be taken account to calculate mip levels. (Mip maps: 32 * 32 * 64, 16 * 16 * 64, 8 * 8 * 64, 4 * 4 * 64, 2 * 2 * 64, 1 * 1 * 64)
+    {
+      width: 32,
+      height: 32,
+      arrayLayerCount: 64,
+      dimension: '3d',
+      mipLevelCount: 7,
+      _success: true,
+    }, // depth of 3D texture should be taken account to calculate mip levels. (Mip maps: 32 * 32 * 64, 16 * 16 * 32, 8 * 8 * 16, 4 * 4 * 8, 2 * 2 * 4, 1 * 1 * 2, 1 * 1 * 1)
   ])
   .fn(async t => {
-    const { width, height, mipLevelCount, _success } = t.params;
+    const { width, height, mipLevelCount, arrayLayerCount, _success } = t.params;
+
+    const descriptor = t.getDescriptor({ width, height, arrayLayerCount, mipLevelCount });
+
+    t.expectValidationError(() => {
+      t.device.createTexture(descriptor);
+    }, !_success);
+  });
 
-    const descriptor = t.getDescriptor({ width, height, mipLevelCount });
+g.test('validation_of_sampleCount')
```

Comment:
uNit: why did this test move? it made the diff a little more verbose.

Reply:
I moved it in order to be consistent to the sequence of TODOs.